### PR TITLE
Feature/activate volunteer

### PR DESCRIPTION
**Pull request by pavlokolodka**
https://trello.com/c/yVisM7lo

### CODE REVIEW

File: src/components/profile/Profile.tsx [hunk 1]
```diff
@@ -23,6 +25,16 @@ export const Profile: FC<IProfileProps> = ({
   orders,
 }) => {
   const navigate = useNavigate();
+  const [user, setUser] = useState<UserType>();
+
+  useEffect(() => {
+    const getUser = async () => {
```
Comment (BillyAleks) [inline src/components/profile/Profile.tsx:31]: Better to move out function from useEffect and just invoke it inside

File: src/components/profile/profile-activate/ProfileActivate.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+import * as yup from 'yup';
+import TextField from '@material-ui/core/TextField';
+import { FC, useEffect, useState } from 'react';
+import { Box, Button, Typography } from '@mui/material';
+import userService from '../../../api/user.service';
+import formStyles from '../../../styles/form-styles';
+import inputStyles from '../../../styles/input-styles';
+import { useNavigate } from 'react-router-dom';
+import { UserType } from '../../../types/auth.types';
+
```
Comment (BillyAleks) [inline src/components/profile/profile-activate/ProfileActivate.tsx:11]: To many blank lines. Normally you should use 1 and just sometimes in rare cases you can use 2 maximum.
It's not a rule in general, but 90% of code base is written in such a behavior

File: src/components/profile/profile-activate/ProfileActivate.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+
+const toBase64 = async (file: Blob): Promise<string> => await new Promise((resolve, reject) => {
```
Comment (BillyAleks) [inline src/components/profile/profile-activate/ProfileActivate.tsx:14]: I would move this function to global helpers, as this function can be useful not only for this component.

File: src/components/profile/profile-activate/ProfileActivate.tsx [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+      .required('Country is required'),
+    city: yup
+      .string()
+      .min(3, 'City should be of minimum 8 characters length')
+      .required('City is required'),
+  });
+        
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    async function getUser(token: string | null) {
```
Comment (BillyAleks) [inline src/components/profile/profile-activate/ProfileActivate.tsx:57]: same about function. Leave only invoking in here

File: src/components/profile/profile-activate/ProfileActivate.tsx [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        justifyContent: 'center',
+      }}
+    >
+      <Box sx={formStyles}>
+        <Typography sx={{ textAlign: 'center', fontSize: '2rem' }}>
+        Activate Profile
```
Comment (BillyAleks) [inline src/components/profile/profile-activate/ProfileActivate.tsx:84]: Indent is missing

File: src/components/profile/profile-activate/ProfileActivate.tsx [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+        </Typography>
+
+        <Formik
+          initialValues={initialValues}
+          validationSchema={validationSchema}
+          onSubmit={async (values, formikHelpers) => {
+            await submitRequest(values, formikHelpers);
+          }}
+        >
+          {({ values, errors, touched, isValid, dirty, setFieldValue }) => {
```
Comment (BillyAleks) [inline src/components/profile/profile-activate/ProfileActivate.tsx:94]: You can return directly JSX here

File: src/components/profile/profile-activate/ProfileActivate.tsx [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+                  id='raised-button-file'
+                  multiple
+                  type='file'
+                  name='document'
+                  onChange={(event) => {
+                    if (!event.currentTarget.files) return;
+                    
+                    const type = event.currentTarget.files[0].type.split('/')[1];
+
+                    setFieldValue('expansion', type);
```
Comment (BillyAleks) [inline src/components/profile/profile-activate/ProfileActivate.tsx:150]: I don't like here that it's potentially two same setFieldValue will trigger re-rendering twice. Is it possible to refactor it to set this value only once?

File: src/pages/profile/profile-activate-page.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import { useNavigate } from 'react-router-dom';
+import ProfileUpdate from '../../components/profile/profile-activate';
+
+const ProfileActivatePage: FC = () => {
+  const [user, setUser] = useState<UserType>();
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const getUser = async () => {
```
Comment (BillyAleks) [inline src/pages/profile/profile-activate-page.tsx:13]: It looks like you have used this function before. What do you think of making it more global and reusable?

File: src/pages/profile/profile-activate-page.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    };
+
+    if (localStorage.getItem('token')) {
+      getUser();
+    } else {
+      navigate('/sign-in');
+    }
+  }, []);
+
+  return (
```
Comment (BillyAleks) [inline src/pages/profile/profile-activate-page.tsx:25]:
```suggestion
if (!user) return null;
return <ProfileUpdate />;
```

**Review by BillyAleks — CHANGES_REQUESTED** [review]
Thanks, please check comments.
I would recommend not to use a blank lines in JSX. It makes it a little bit harder to see the hierarchy.

**Comment by pavlokolodka** [comment]
Merge with https://github.com/CHI-internship/KraudDonate-Service/pull/36

**Review by lerik1408 — APPROVED** [review]

**Pull request merged**